node-zerox/src/index.ts

```diff
@@ -11,6 +11,28 @@ import fs from "fs-extra";
 import os from "os";
 import path from "path";
 import pLimit, { Limit } from "p-limit";
+import { LLMParams } from "./types";
+
+const defaultLLMParams: LLMParams = {
+  maxTokens: 1000,
+  temperature: 0,
+  topP: 1, // OpenAI defualts to 1
+  frequencyPenalty: 0, // OpenAI default to 0
+  presencePenalty: 0, // OpenAI default to 0
+};
+
+const validateLLMParams = (params: Partial<LLMParams>): LLMParams => {
+  const validKeys = Object.keys(defaultLLMParams);
+  const invalidKeys = Object.keys(params).filter(
+    (key) => !validKeys.includes(key)
+  );
+
```

Comment:
Maybe we should check if the value is correct type of number. Right now, it seems I can pass in i.e. topP: [], which causes OpenAI to throw an error. What are your thoughts?

Reply:
If we want to validate the type of the value as well, then either manually validate (switch case) or Zod. Without Zod basically have to do type gymnastics manually.